topics/variant-analysis/tutorials/microbial-variants/tutorial.md

```diff
@@ -0,0 +1,235 @@
+> <br/>
+> <br/>
+> Your tool interface should look like this:
+>
+>  ![Snippy interface](../../images/interface.png)
+>
+> - Click **Execute**.
+>
+{: .hands_on}
+
+## Step 3: Examine Snippy output
+
+Snippy has taken the reads, mapped them against the reference using BWA MEM, looked through the resulting BAM file and found differences using some fancy Bayesian statistics (Freebayes), filtered the differences for sensibility and finally checked what effect these differences will have on the predicted genes and other features in the genome.
+
+It produces quite a bit of output, there can be up to 10 output files.
+
+| Filename | Description  |
+|----------|--------------|
+| snps vcf file | The final annotated variants in VCF format  |
+|----------|--------------|
+| snps gff file | The variants in GFF3 format |
+|----------|--------------|
+| snps table | A simple tab-separated summary of all the variants |
+|------------|------------|
+| snps summary  | A summary of the snps called  |
+|------------|------------|
+| log file  | A log file with the commands run and their outputs  |
+|------------|------------|
+| aligned fasta | A version of the reference but with - at position with depth=0 and N for 0 < depth < --mincov **(does not have variants)**  |
+|------------|------------|
+| consensus fasta | A version of the reference genome with all variants instantiated  |
+|------------|------------|
+| mapping depth | A table of the mapping depth  |
+|------------|------------|
+| mapped reads bam  | A bam file containing all of the mapped reads |
+|------------|------------|
+| outdir  | A tarball of the Snippy output directory for inout into Snippy-core if required |
+|------------|------------|
+
+We will now have a look at the contents of the snp table file.
+
+> ### {% icon hands_on %} Hands-on: View the SNPs
+>
+> 1. Go to the file called *snippy on data XX, data XX and data XX table* and click on the eye icon.
+>   - We can see a list of variants. Look in column 3 to see which types the variants are, such as a SNP or a deletion.
+> 1. Look at the third variant called. This is a T&rarr;A mutation, causing a stop codon. Look at column 14: the product of this gene is a methicillin resistance protein. Methicillin is an antibiotic.
+>
+>    > ### :question: Question
+>    >
+>    > What might be the result of such a mutation?
+>    >
+>    > <details>
+>    > <summary>Click to view answers</summary>
+>    > This will cause a truncation in the Methicillin gene and a loss of resistance in the organism.
+>    > </details>
+>    {: .question}
+{: .hands_on}
+
+# Step 3: View Snippy output in JBrowse
+
+We could go through all of the variants like this and read them out of a text table, but this is onerous and doesn't really give the context of the changes very well. It would be much nicer to have a visualisation of the snps and the other relevant data. In Galaxy we can use a tool called JBrowse.
+
+> ### {% icon hands_on %} Hands-on: Run JBrowse
```

Comment:
I tried to do the same as for the first hands-on and merged the two next hands-on sections:

```
> ### {% icon hands_on %} Hands-on: Inspecting the SNPs using JBrowse
>
> 1. Search for **JBrowse** {% icon tool %} and run it with the following parameters
>    - "Reference genome to display" to `Use a genome from history`
>    - "Select the reference genome" to `wildtype.fna`
>       
>       This sequence will be the reference against which annotations are displayed
>
>    - "Produce Standalone Instance" to `Yes`
>    - "Genetic Code" to `11: The Bacterial, Archaeal and Plant Plastid Code`
>    - "JBrowse-in-Galaxy Action" to `New JBrowse Instance`
>    - "Track Group"
>       
        We will now set up three different tracks - these are datasets displayed underneath the reference sequence (which is displayed as nucleotides in FASTA format). We will choose to display the sequence reads (the .bam file), the variants found by snippy (the .gff file) and the annotated reference genome (the wildtype.gff)
>
>       - **Track 1 - sequence reads**: Click on `Insert Track Group` and fill it with
>           - "Track Cateogry" to `sequence reads`
>           - Click on `Insert Annotation Track` and fill it with
>               - "Track Type" to `BAM Pileups`
>               - "BAM Track Data" to `snippy bam file`
>               - "Autogenerate SNP Track" to `Yes`
>               - "Track Visibility" to `On for new users`
>       - **Track 2 - variants**: Click on `Insert Track Group` and fill it with
>           - "Track Cateogry" to `variants`
>           - Click on `Insert Annotation Track` and fill it with
>               - "Track Type" to `GFF/GFF3/BED/GBK Features`
>               - "GFF/GFF3/BED Track Data" to `snippy snps gff file`
>               - "Track Visibility" to `On for new users`
>       - **Track 3 - annotated reference**: Click on `Insert Track Group` and fill it with
>           - "Track Cateogry" to `annotated reference`
>           - Click on `Insert Annotation Track` and fill it with
>               - "Track Type" to `GFF/GFF3/BED/GBK Features`
>               - "GFF/GFF3/BED Track Data" to `wildtype.gff`
>               - "Track Visibility" to `On for new users`
>               - "JBrowse Track Type [Advanced]" to `Canvas Features`
>               - Click on "JBrowse Styling Options [Advanced]"
>               - "JBrowse style.label" to `product`
>               - "JBrowse style.description" to `product`
>               - "Track Visibility" to `On for new users`
>
>    A new file will be created in your history, this contains the JBrowse interactive visualisation. We will now view it's contents and play with it
> 
> 2. Inspect the `JBrowse on data XX and data XX - Complete` file by clicking on the eye icon
>
>    The JBrowse window will appear in the centre Galaxy panel.
>
> 3. Display all the tracks and practice maneuvering around
>    1. Click on the tick boxes on the left to display the tracks
>    1. Zoom out by clicking on the `minus` button to see sequence reads and their coverage (the grey graph)
>    1. Zoom in by clicking on the `plus` button to see
>       - probable real variants (a whole column of snps)
>       - probable errors (single one here and there)
>
>    ![JBrowse screenshot](../../images/jbrowse1.png "Screenshot of JBrowse")
>
> 4. Look at the stop SNP
>    1. Type `47299` in the coordinates box
>    2. Click on `Go` to see the position of the SNP discussed above
>   
>    The correct codon at this position is TGT, coding for the amino acid Cysteine (middle row of the amino acid translations). The mutation of T &rarr; A turns this triplet into TGA, a stop codon.
>
>    ![JBrowse screenshot](../../images/jbrowse2.png "Inspection of the STOP SNP using JBrowse")
>
{: .hands_on}
```
It gives sth like:
![screen shot 2018-01-23 at 15 00 11](https://user-images.githubusercontent.com/1842467/35279724-2c15063a-004e-11e8-8a46-2135844f436e.png)
![screen shot 2018-01-23 at 15 00 23](https://user-images.githubusercontent.com/1842467/35279725-304ba9fc-004e-11e8-977c-71801d392a59.png)

What do you think?